addons/controllers/addon_controller.go

```diff
@@ -234,6 +234,9 @@ func (r *AddonReconciler) reconcileDelete(
 		return ctrl.Result{}, kerrors.NewAggregate(errors)
 	}
 
+	if controllerutil.ContainsFinalizer(cluster, addontypes.AddonFinalizer) {
+		controllerutil.RemoveFinalizer(cluster, addontypes.AddonFinalizer)
```

Comment:
Does this call update/patch the object. I dont think it does that.

Reply:
fixed. added the call to patch in new version of pr